pkg/scalers/mongo.go

```diff
@@ -0,0 +1,291 @@
+			meta.password = v
+		} else if v, ok := config.TriggerMetadata["passwordFromEnv"]; ok {
+			meta.password = config.ResolvedEnv[v]
+		}
+
+		if len(meta.password) == 0 {
+			return nil, "", fmt.Errorf("no password given")
+		}
+	}
+
+	if meta.connectionString != "" {
+		connStr = meta.connectionString
+	} else {
+		// Build connection str
+		addr := fmt.Sprintf("%s:%s", meta.host, meta.port)
+		auth := fmt.Sprintf("%s:%s", meta.username, meta.password)
+		connStr = "mongodb://" + auth + "@" + addr
+	}
+
+	return &meta, connStr, nil
+}
+
+func (s *mongoScaler) IsActive(ctx context.Context) (bool, error) {
+	result, err := s.getQueryResult()
+	if err != nil {
+		mongoLog.Error(err, fmt.Sprintf("failed to get query result by mongo,because of %v", err))
+		return false, err
+	}
+	return result > 0, nil
+}
+
+// Close disposes of mongo connections
+func (s *mongoScaler) Close() error {
+	if s.client != nil {
+		err := s.client.Disconnect(context.TODO())
+		if err != nil {
+			mongoLog.Error(err, fmt.Sprintf("failed to close mongo connection,because of %v", err))
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getQueryResult query mongo by meta.query
+func (s *mongoScaler) getQueryResult() (int, error) {
+	var (
+		results []mongoFields
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), mongoDefaultTimeOut)
+	defer cancel()
+
+	filter, err := json2BsonDoc(s.metadata.query)
+	if err != nil {
+		mongoLog.Error(err, fmt.Sprintf("failed to convert query param to bson.Doc,because of %v", err))
+		return 0, err
+	}
+
+	cursor, err := s.client.Database(s.metadata.dbName).Collection(s.metadata.collection).Find(ctx, filter)
+	if err != nil {
+		mongoLog.Error(err, fmt.Sprintf("failed to query %v in %v,because of %v", s.metadata.dbName, s.metadata.collection, err))
+		return 0, err
+	}
+
+	err = cursor.All(ctx, &results)
+	if err != nil {
+		mongoLog.Error(err, fmt.Sprintf("failed to traversal the query results,because of %v", err))
+		return 0, err
+	}
+
+	return len(results), nil
+}
+
+// GetMetrics query from mongo,and return to external metrics
+func (s *mongoScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	num, err := s.getQueryResult()
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("failed to inspect momgo,because of %v", err)
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(num), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+// GetMetricSpecForScaling get the query value for scaling
+func (s *mongoScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetQueryValue := resource.NewQuantity(int64(s.metadata.queryValue), resource.DecimalSI)
+	metricName := kedautil.NormalizeString(fmt.Sprintf("%s-%s", "mongo", s.metadata.connectionString))
```

Comment:
If there are multiple mongodb triggers specified in one ScaledObject, we need to be sure, that `metricName` used for each trigger is unique. If more mongo triggers will use the same connection string, the `metricNames` will be same. We need to avoid that, please refer to PostrgreSQL or InfluxDB scaler on how to achieve that (in short we do the best effort to generate an unique metric name, but we allow users to override this value with an optional parameter).

Docs:
https://keda.sh/docs/2.1/scalers/postgresql/

https://keda.sh/docs/2.1/scalers/influxdb/